plugins/out_datadog/datadog_remap.c

```diff
@@ -28,172 +28,98 @@ const char *ECS_ARN_PREFIX = "arn:aws:ecs:";
 const char *ECS_CLUSTER_PREFIX = "cluster/";
 const char *ECS_TASK_PREFIX = "task/";
 
-static int dd_remap_append_kv_to_ddtags(const char *key,
-                                        const char *val, size_t val_len, flb_sds_t *dd_tags_buf)
+static void dd_remap_append_kv_to_ddtags(const char *key,
+                                         const char *val, size_t val_len, flb_sds_t dd_tags)
 {
-    flb_sds_t tmp;
-
-    if (flb_sds_len(*dd_tags_buf) != 0) {
-        tmp = flb_sds_cat(*dd_tags_buf, FLB_DATADOG_TAG_SEPERATOR, strlen(FLB_DATADOG_TAG_SEPERATOR));
-        if (!tmp) {
-            flb_errno();
-            return -1;
-        }
-        *dd_tags_buf = tmp;
+    if (flb_sds_len(dd_tags) != 0) {
+        flb_sds_cat(dd_tags, FLB_DATADOG_TAG_SEPERATOR, strlen(FLB_DATADOG_TAG_SEPERATOR));
     }
-
-    tmp = flb_sds_cat(*dd_tags_buf, key, strlen(key));
-    if (!tmp) {
-            flb_errno();
-            return -1;
-        }
-    *dd_tags_buf = tmp;
-
-    tmp = flb_sds_cat(*dd_tags_buf, ":", 1);
-    if (!tmp) {
-            flb_errno();
-            return -1;
-        }
-    *dd_tags_buf = tmp;
-
-    tmp = flb_sds_cat(*dd_tags_buf, val, val_len);
-    if (!tmp) {
-            flb_errno();
-            return -1;
-        }
-    *dd_tags_buf = tmp;
-
-    return 0;
+    flb_sds_cat(dd_tags, key, strlen(key));
+    flb_sds_cat(dd_tags, ":", 1);
+    flb_sds_cat(dd_tags, val, val_len);
 }
 
 /* default remapping: just move the key/val pair under dd_tags */
-static int dd_remap_move_to_tags(const char *tag_name,
-                                  msgpack_object attr_value, flb_sds_t *dd_tags_buf)
+static void dd_remap_move_to_tags(const char *tag_name,
+                                  msgpack_object attr_value, flb_sds_t dd_tags)
 {
-    return dd_remap_append_kv_to_ddtags(tag_name, attr_value.via.str.ptr,
-                                        attr_value.via.str.size, dd_tags_buf);
+    dd_remap_append_kv_to_ddtags(tag_name, attr_value.via.str.ptr,
+                                 attr_value.via.str.size, dd_tags);
 }
 
 /* remapping function for container_name */
-static int dd_remap_container_name(const char *tag_name,
-                                    msgpack_object attr_value, flb_sds_t *dd_tags_buf)
+static void dd_remap_container_name(const char *tag_name,
+                                    msgpack_object attr_value, flb_sds_t dd_tags)
```

Comment:
help me understand again, why are we reverting these changes which probably still are a real fix? Also if we are going to revert them, why not put it in a revert commit? Why same commit as this new fix.